lib/bolt/plugin/vault.rb

```diff
@@ -0,0 +1,185 @@
+# frozen_string_literal: true
+
+module Bolt
+  class Plugin
+    class Vault
+      class VaultHTTPError < Bolt::Error
+        def initialize(response)
+          err = JSON.parse(response.body)['errors']
+          m = String.new("#{response.code} \"#{response.msg}\"")
+          m << ": #{err.join(';')}" unless err.nil?
+          super(m, 'bolt.plugin/vault-http-error')
+        end
+      end
+
+      attr_reader :config
+
+      # All requests for secrets must have a token in the request header
+      TOKEN_HEADER = "X-Vault-Token"
+
+      # Default header for all requests, including auth methods
+      DEFAULT_HEADER = {
+        "Content-Type" => "application/json",
+        "Accept" => "application/json"
+      }.freeze
+
+      # Make sure no unexpected keys are in the config
+      def validate_config(config)
+        known_keys = %w[server_url auth]
+
+        config.each do |key, _v|
+          next if known_keys.include?(key)
+          raise Bolt::ValidationError, "Unexpected key in Vault plugin config: #{key}"
+        end
+      end
+
+      # Make sure no unexpected keys are in the inventory config and
+      # that required keys are present
+      def validate_options(opts)
+        known_keys = %w[_plugin server_url auth path field version]
+        required_keys = %w[path]
+
+        opts.each do |key, _v|
+          next if known_keys.include?(key)
+          raise Bolt::ValidationError, "Unpexpected key in inventory config: #{key}"
+        end
+
+        required_keys.each do |key|
+          next if opts[key]
+          raise Bolt::ValidationError, "Expected key in inventory config: #{key}"
+        end
+      end
+
+      def name
+        'vault'
+      end
+
+      def hooks
+        ['inventory_config']
+      end
+
+      def initialize(config)
+        validate_config(config)
+        @config = config
+        @logger = Logging.logger[self]
+      end
+
+      def inventory_config(opts)
+        validate_options(opts)
+
+        header = {
+          TOKEN_HEADER => token(opts)
+        }
+
+        response = request(:Get, uri(opts), nil, header)
+
+        parse_response(response, opts)
+      end
+
+      # Request uri - built up from Vault server url and secrets path
+      def uri(opts, path = nil)
+        url = opts['server_url'] || config['server_url'] || ENV['VAULT_ADDR']
+
+        # Handle the different versions of the API
+        if opts['version'] == 2
+          opts['path'] = opts['path'].split('/').insert(1, 'data').join('/')
```

Comment:
This could also be something like:

```rb
mount, key = opts['path'].split('/', 2)
opts['path'] = [mount, 'data', key].join('/')
```

`split('/', 2)` will split it into exactly two parts, preserving slashes in the second part. Might be clearer than insert, but I'll leave that up to you.